design/one-pager-function-go-templating.md

```diff
@@ -0,0 +1,171 @@
+# Go Templating Function
+ 
+* Owner: Ezgi Demirel (@ezgidemirel)
+* Reviewers: Crossplane Maintainers
+* Status: Accepted
+
+## Background
+
+With Crossplane 1.14, Composition Functions will be promoted to `v1beta1`, and several 
+composition functions will be introduced as an extension point to be used in pipelines. 
+One of these extensions is the [Go Templating Function], which allows users to render Crossplane
+resources using Go templating capabilities like conditionals and loops. In addition to that, 
+users can use values in the environment configs or other resource fields to create/configure
```

Comment:
Is there an example somewhere how to do it? If I understand correctly, this looks like the _values_ file in helm. How is it different from the `environmentConfigs`?

Reply:
So those are actual environmental variables that the pod has access to as opposite to the `environmentConfig` file?

Reply:
No, they're `environmentConfig` variables. Maybe naming the function as `getEnvConfigValue` makes more sense.

Reply:
Thank you for clarifying, I need to catch up on the `environmentConfig` design :)